cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -453,19 +453,27 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
         },
 
         getRequestData: function() {
+            var sectionIds = $('.outline-section').map(function(){return this.id;}).get()
+            var subsectionIds = $('.outline-subsection').map(function(){return this.id;}).get()
+            var relative_weeks_due = null;
             if (this.getValue() < 19 && this.getValue() > 0 && $('#grading_type').val() !== 'notgraded') {
-                return {
-                    metadata: {
-                        relative_weeks_due: this.getValue()
-                    }
-                };
-            } else {
-                return {
-                    metadata: {
-                        relative_weeks_due: null
-                    }
-                }
+                relative_weeks_due = this.getValue()
             }
+            window.analytics.track('edx.bi.studio.relative_date.saved', {
+                block_id: this.model.get('id'),
+                courserun_key: course.get('id'),
+                num_of_sections_in_course: $('.outline-section').length,
+                num_of_subsections_in_course: $('.outline-subsection').length,
+                order_in_sections: sectionIds.indexOf(this.parent.options.parentInfo.get('id')) + 1,
+                order_in_subsections: subsectionIds.indexOf(this.model.get('id')) + 1,
+                org_key: course.get('org'),
+                relative_weeks_due: relative_weeks_due,
+            });
+            return {
+                metadata: {
+                    relative_weeks_due: relative_weeks_due
```

Comment:
Not sure if it works here and you definitely **do not** need to change, but I _think_ you might be able to only include `relative_weeks_due` and JS is smart enough to know that the key and value use the same name.
So basically
```
{
  relative_weeks_due
}
```
===
```
{
  relative_weeks_due: relative_weeks_due
}
```

This is more just an FYI though for your future endeavors